staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go

```diff
@@ -275,6 +277,18 @@ func (az *Cloud) DeletePublicIP(service *v1.Service, pipResourceGroup string, pi
 	if rerr != nil {
 		klog.Errorf("PublicIPAddressesClient.Delete(%s) failed: %s", pipName, rerr.Error().Error())
 		az.Event(service, v1.EventTypeWarning, "DeletePublicIPAddress", rerr.Error().Error())
+
+		errMsgCode, err := extractRetryErrorMsgCode(rerr.Error().Error())
+		if err != nil {
+			klog.Errorf("DeletePublicIP for public IP %s failed when extracting error message code with error %s", pipName, err.Error())
+			return err
+		}
+
+		if strings.EqualFold(cannotDeletePublicIPErrorMessageCode, errMsgCode) {
+			klog.Warningf("DeletePublicIP for public IP %s failed with error code PublicIPAddressCannotBeDeleted, this is because other resources are referencing the public IP. The deletion of the service will continue.", pipName)
+			return nil
+		}
+
```

Comment:
I want to ensure that the service can be deleted when there is a problem blocking the public IP deletion, e.g., an inbound NAT rule is referencing the frontend IP configuration. In the current logic, the frontend IP config could not be deleted, leading to the failure of the public IP deletion operation. Here I filter out this kind of error to make sure the service is deleted. In this case, the frontend IP configuration may not be deleted automatically. It is considered that the users delete it manually because the azure provider only focuses on deleting the frontend IP config which belongs to the service. Hence, if the service is gone and the frontend IP config fails to be deleted because of the references of other resources, azure cannot delete it as well.

Reply:
this is likely another bug fix? it's better fix it independently, e.g. we may cherry-pick this fix to old realease.